### ISSUE TITLE
This option as a toggle: `Edit->Node core->Edit node core in-line`

### ISSUE THREAD

**Issue by euu2021** [freeplane/freeplane#1279]
Currently, applying the `Edit->Node core->Edit node core in-line` into a node that is already with the "edit in-line" opened, does nothing. I suggest that, instead, it closes the "edit in-line" mode.

My situation: I use minimized nodes a lot, and I navigate them in 2 ways:

1.  with the mouse: hovering with the mouse, which triggers the "text preview window"
2.  with the keyboard: opening each node using `Edit->Node core->Edit node core in-line`

When navigating with this second method, it is a 3 steps process:

1. open the node using the hotkey for `Edit->Node core->Edit node core in-line`
2. read
3. close the node using `Enter`

So, I need to press one button to open the node, and a different button to close the node. I would like to have the open and close set to the same hotkey, so I don't need to move my hand back and forth between those 2 buttons. In other words, a toggle. 

This would significantly improve my experience, considering that I need to go through thousands of nodes doing this 3 steps open and close process.

Thanks,

**Comment by euu2021** [comment]
Ok, I found a funny workaround: setting `Esc` as the hotkey for `Edit->Node core->Edit node core in-line`, it starts working as a toggle. 

I keep the request, though, because it will be impossible to convince my brain that Esc is not "Go to Root" anymore 🤣🤣

PS: this workaround fails also because, if I do any edits in the text, closing with Esc cancels those edits.

**Comment by amariusz** [comment]
Why not using Alt+F3 (Minimize note) as a toggle?

> When navigating with this second method, it is a 3 steps process:
> 
> 1. open the node using the hotkey for `Edit->Node core->Edit node core in-line`
> 2. read
> 3. close the node using `Enter`

**Comment by euu2021** [comment]
> Why not using Alt+F3 (Minimize note) as a toggle?

Good point. To explain, I will give more details about my usage:

As I said above, I navigate either with the keyboard or with the mouse. 

I go with the mouse if I'm in a moment when I will just read the nodes. Simply hovering the mouse is much faster than pressing buttons to open and close each node.

I go with the keyboard when I'm more prone to make changes in the nodes. Because of that, I open them with `Edit->Node core->Edit node core in-line`, so, if there is an edit to do, I'm already in "edit in line mode". 

So, using `View-Minimize node` does not fit my situation.

**Comment by amariusz** [comment]
I agree it would be nice to have an ability to assign shortcuts for edit mode and that you could use it as a toggle.

However in this case I can't wrap my head around why two keys are not convenient enough.

Assuming you have standard or TKL keyboard :
- Without moving your left hand you can press F2 (edit) and Esc (exit). You use cursors for navigation or mouse.
- Without much movement of your right hand you can navigate with cursors, enter edit mode with End (or Home), and exit with Enter.

Another approach - ctrl+a to select all nodes, alt+f3 to minimize or 'unminimize' all nodes at once.

**Comment by euu2021** [comment]
> However in this case I can't wrap my head around why two keys are not convenient enough.

Notice that, 
-if I use Esc to close, the process will involve 3 keys: F2, Esc, Enter
-with my suggestion, it will go down to 1 key: F2

With 1 key, it's much easier on the brain. It's the one key to rule them all.

>    One Key to rule them all, So I don't need to find them,
   One Key to control them all, and in only one hotkey bind them

![image](https://github.com/freeplane/freeplane/assets/77707706/a16f5479-18c9-48f3-a6f7-9733487fd957)


> Another approach - ctrl+a to select all nodes, alt+f3 to minimize or 'unminimize' all nodes at once.

This wouldn't work for me, because I need the nodes minimized. If I 'unminimize' the nodes, the content goes completely out of control. Compare the difference at the same zoom level:

Only one node 'unminimized', I can get a reasonable overview:
![image](https://github.com/freeplane/freeplane/assets/77707706/93a4aea4-0508-40c4-beda-77bcff8197d7)

All nodes 'unminimized', I get lost in the forest:
![image](https://github.com/freeplane/freeplane/assets/77707706/34c025f5-0eb1-4fd6-b88d-d5eb94bbec4a)

**Comment by amariusz** [comment]
OK, I get it :)

**Comment by macmarrum** [comment]
Based on the above and on
- #1280 

I understand that you'd like 
1. to have the option to disable `Enter` as the terminator of in-line editing (`Enter` will do new line)
2. the same hot key as for `Edit->Node core->Edit node core in-line` do the termination in the in-line editor

I'd go one step further, i.e.

|||
|-|-|
|I WANT| Freeplane to allow to specify a hot key for terminating in-line-node editing (by default `Enter`), in addition to the "universal" `Esc`|
|ALTERNATIVELY| Freeplane to have an option to allow the usage of the same hot key for terminating in-line-node editing as for `Edit->Node core->Edit node core in-line`, instead of the default `Enter`, in addition to the "universal" `Esc`|
|SO THAT| `Enter` is freed up to insert a new line and e.g. `F2` can be used instead, which will make the same hot key do `Edit->Node core->Edit node core in-line (F2)` and **terminate in-line editing**|

**Comment by euu2021** [comment]
Yes, good sum up. 

For clarification, I would just add that Esc would remain as the exclusive holder of the role: "negative terminator". Ie, Cancel the changes and close the in-line editor. The other terminators save the changes.

**Comment by amariusz** [comment]
> This wouldn't work for me, because I need the nodes minimized. If I 'unminimize' the nodes, the content goes completely out of control. Compare the difference at the same zoom level:

BTW - one more tip. Looking at your screenshots it seems that lengthy content is mostly located within leaf nodes. In this case you could consider adding a new user style with increased `max node width` and setting it as a conditional style for leaf nodes. This approach should result in better usage of screen real estate by reducing the number of wrapped lines.

**Comment by euu2021** [comment]
> BTW - one more tip. Looking at your screenshots it seems that lengthy content is mostly located within leaf nodes. In this case you could consider adding a new user style with increased `max node width` and setting it as a conditional style for leaf nodes. This approach should result in better usage of screen real estate by reducing the number of wrapped lines.

Thanks for this interesting suggestion. Unfortunately, it doesn't fit my situation. 

For years, I've been following the principle of keeping symmetry between the nodes. This was first proposed years ago by Jonas Batista ([see here](https://sourceforge.net/p/freeplane/discussion/758437/thread/b5ad5263/)). Later, I evolved this principle by proposing the idea of having a vertical symmetry option, so that even longer paragraphs would keep the symmetry (Macmarrum created the main script [here](https://sourceforge.net/p/freeplane/discussion/758437/thread/e9200db6c3)). 

In order for this script approach to work, I need to limit the number of shown characters in the minimized nodes (Preferences…->Appearance->Size limits->Maximum shortened text width) to a number that fits one line. A side effect of this is that, even if I increase the width of the node, no extra content of it is shown. Example (the nodes have more content than what is showed, but FP limits to this number of characters):


![image](https://github.com/freeplane/freeplane/assets/77707706/10360edc-5fea-46f5-9bec-6de801f156e9)



Anyway, I can strongly recommend for people to try this principle of symmetry, because it makes the contents of the map much more manageable. Horizontal symmetry has a vanilla option (make the max and min node width equal). For vertical symmetry, you need the script workaround (it has some limitations).

PS for clarification: to sum up, the suggestion doesn't fit my needs because:

1. I like to keep horizontal symmetry
2. It's not technically possible

**Comment by amariusz** [comment]
Thanks for the explanation and links. I think I understand now why your mindmap has this specific "column look". 

The original video is no longer available but as far as I understand this is the same concept:

> Live Freeplane Mind Mapping - The Power of a Mind to Map: Tony Buzan at TEDxSquareMile - YouTube - https://www.youtube.com/watch?v=-9nldrnZMig

IMO it definitely has application to many cases - especially if you work with mind maps that are more structured than freeform. But I also find this comment on point:
https://sourceforge.net/p/freeplane/discussion/758437/thread/b5ad5263/#0006/0b06/2108/5cf5

Anyway, two thoughts come into my mind:

1. It seems to me that the option `Maximum shortened text width` could be splitted into two in Freeplane - one for leaf nodes and one for non-leaf nodes. This way minimized leaf nodes could be much longer/wider which seems sensible to me as usually (at least in my maps) they convey most of the contents. That would of course require changes to your scripts that manage `minimized` state to pick the right setting depending on `isLeaf` property.
2. I personally prefer to insert newline in a node core text to explicitly determine 'minimized' contents even if this results in one or two line wraps. Then `Maximum shortened text width` can be further used to force shortened form if required.

So maybe with above changes it would be technically possible to keep horizontal symmetry for non-leaf nodes and at the same time have readable leaf-nodes that make better use of space available?

**Comment by amariusz** [comment]
Couple more thoughts after sleeping on it ;)

> 1. It seems to me that the option `Maximum shortened text width` could be splitted into two in Freeplane - one for leaf nodes and one for non-leaf nodes.

Alternative approach could be implementing `Maximum shortened text width` setting at node *style* level with default  value (used in "Default" node style) taken from global preferences (`Maximum shortened text width`). 

This would be a more flexible solution. Longer minimized leaf nodes could be achieved as one of many possible scenarios. Another benefit would be the possibility to have this setting specific to mind map files, not only global application setting.

Regarding some of the scripts you use:
https://github.com/euu2021/FreePlane-Scripts/blob/main/scripts/Auto%20Minimize/paste_and_minimize.groovy
https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/minimizeAllNodesIfTextIsLonger.groovy
(BTW wonderful collection @macmarrum !)

Instead of using `config.getIntProperty`, scripts could (somehow..) check the NodeStyle attribute for `Maximum shortened text width` (similarly to fontSize or border width).

Also,  in such case maybe the `Defaults` tab in Preferences would be better fit for `Maximum shortened text width` then `Appearance` tab?

@dpolivaev - what do you think about such change in FreePlane? Maybe it was already disscussed in the past?
If that's something of interest it should probably be moved to separate thread.

**Comment by euu2021** [comment]
I will answer on a new thread, to avoid derealing the current thread too much

**Comment by amariusz** [comment]
Good idea. Regarding replies to`Maximum shortened text width` please use  ~~#1292.~~

Please use: #1293
